iOS/DuckDuckGo/DuckPlayer/NativeUI/Views/DuckPlayerContainer.swift

```diff
@@ -122,55 +141,123 @@ private struct SheetView<Content: View>: View {
     let containerHeight: Double
     let content: (DuckPlayerContainer.PresentationMetrics) -> Content
     let onHeightChange: (Double) -> Void
+    let onDismiss: () -> Void
+    let onPresentDuckPlayer: () -> Void
 
     @State private var sheetHeight: Double = 0
     @State private var sheetWidth: Double?
     @State private var opacity: Double = 0
-    @State private var sheetOffset = DuckPlayerContainer.Constants.Offset.initialValue
+    @State private var sheetOffset = DuckPlayerContainer.Constants.initialOffsetValue
+    @GestureState private var dragStartOffset: Double?
+    @State private var isDragging = false
+    @State private var isAnimatingToTop = false
 
     // Animate the sheet offset with a spring animation
     private func animateOffset(to visible: Bool) {
-
+        let offset = calculateSheetOffset(for: visible, containerHeight: containerHeight)
         if #available(iOS 17.0, *) {
             withAnimation(
                 .spring(
-                    duration: DuckPlayerContainer.Constants.Animation.springDuration, bounce: DuckPlayerContainer.Constants.Animation.springBounce)
+                    duration: DuckPlayerContainer.Constants.springDuration, bounce: DuckPlayerContainer.Constants.springBounce)
             ) {
-                sheetOffset = calculateSheetOffset(for: visible, containerHeight: containerHeight)
+                sheetOffset = offset
             } completion: {
                 viewModel.sheetAnimationCompleted = true
             }
         } else {
             withAnimation(
                 .spring(
-                    duration: DuckPlayerContainer.Constants.Animation.springDuration, bounce: DuckPlayerContainer.Constants.Animation.springBounce)
+                    duration: DuckPlayerContainer.Constants.springDuration, bounce: DuckPlayerContainer.Constants.springBounce)
             ) {
-                sheetOffset = calculateSheetOffset(for: visible, containerHeight: containerHeight)
+                sheetOffset = offset
             }
-            // For earlier iOS versions, set completed immediately
             viewModel.sheetAnimationCompleted = true
         }
     }
 
     var body: some View {
-        VStack(alignment: .center) {
-
+        VStack(alignment: .center, spacing: 0) {
             if let sheetWidth {
-                content(DuckPlayerContainer.PresentationMetrics(contentWidth: sheetWidth))
+                VStack(spacing: 0) {
```

Comment:
nit: guard would probably be better here now that the function is bigger

Reply:
ah, yes.  You can't guard on ViewBuilder